connect/transforms/src/main/java/org/apache/kafka/connect/transforms/MaskField.java

```diff
@@ -0,0 +1,173 @@
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.connect.transforms;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.util.SimpleConfig;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.kafka.connect.transforms.util.Requirements.requireMap;
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+public abstract class MaskField<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    public static final String OVERVIEW_DOC =
+            "Mask specified fields with a valid null value for the field type (i.e. 0, false, empty string, and so on)."
+                    + "<p/>Use the concrete transformation type designed for the record key (<code>" + Key.class.getCanonicalName() + "</code>) "
+                    + "or value (<code>" + Value.class.getCanonicalName() + "</code>).";
+
+    private static final String FIELDS_CONFIG = "fields";
+
+    public static final ConfigDef CONFIG_DEF = new ConfigDef()
+            .define(FIELDS_CONFIG, ConfigDef.Type.LIST, "", ConfigDef.Importance.HIGH, "Names of fields to mask.");
+
+    private static final String PURPOSE = "mask fields";
+
+    private static final Map<String, Object> PRIMITIVE_VALUE_MAPPING = new HashMap<>();
+
+    static {
+        PRIMITIVE_VALUE_MAPPING.put(Boolean.class.getCanonicalName(), Boolean.FALSE);
+        PRIMITIVE_VALUE_MAPPING.put(Byte.class.getCanonicalName(), (byte) 0);
+        PRIMITIVE_VALUE_MAPPING.put(Short.class.getCanonicalName(), (short) 0);
+        PRIMITIVE_VALUE_MAPPING.put(Integer.class.getCanonicalName(), 0);
+        PRIMITIVE_VALUE_MAPPING.put(Long.class.getCanonicalName(), 0L);
+        PRIMITIVE_VALUE_MAPPING.put(Float.class.getCanonicalName(), 0f);
+        PRIMITIVE_VALUE_MAPPING.put(Double.class.getCanonicalName(), 0d);
+        PRIMITIVE_VALUE_MAPPING.put(BigInteger.class.getCanonicalName(), BigInteger.ZERO);
+        PRIMITIVE_VALUE_MAPPING.put(BigDecimal.class.getCanonicalName(), BigDecimal.ZERO);
+        PRIMITIVE_VALUE_MAPPING.put(Date.class.getCanonicalName(), new Date(0));
+        PRIMITIVE_VALUE_MAPPING.put(String.class.getCanonicalName(), "");
```

Comment:
Not for the Java type `Byte`, I mean for the Connect `BYTES` schema type, i.e. `byte[].class`